src/config.rs

```diff
@@ -142,6 +142,22 @@ impl Config {
             let parsed_value = serde_json::from_str(&value)
                 .unwrap_or_else(|_| serde_json::Value::String(value.to_string()));
 
+            if key == "book" || key == "build" {
+                if let serde_json::Value::Object(ref map) = parsed_value {
+                    // To `set` each `key`, we wrap them as `prefix.key`
+                    let prefix = &key; // "book" or "build"
+                    let mut s = String::new();
+                    for (k, v) in map {
+                        s.clear();
+                        s.push_str(prefix);
+                        s.push('.');
+                        s.push_str(k);
+                        self.set(&s, v).expect("unreachable");
+                    }
```

Comment:
I think it would be better to simplify, maybe something this like:

```suggestion
                    for (k, v) in map {
                        let full_key = format!("{}.{}", key, k);
                        self.set(&full_key, v).expect("unreachable");
                    }
```

Reply:
Thank you very much. I tried to reduce the string allocation, but simplicity may be more important in this case. I'll try commiting your suggestion.